src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +373,145 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
+				Movement movement = new Movement(medical, chargeType, null, currentLot, now, ajustQty.intValue(), supplier, chargeReferenceNumber);
+				chargeMovements.add(movement);
+			} else if (ajustQty < 0) { // discharge movement when realQty < theoQty
+				Movement movement = new Movement(medical, dischargeType, ward, currentLot, now, -ajustQty.intValue(), null, dischargeReferenceNumber);
+				dischargeMovements.add(movement);
+			} // else ajustQty = 0, continue
+		}
+		// create movements
+		List<Movement> insertedMovements = new ArrayList<>();
+		if (!chargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleChargingMovements(chargeMovements, chargeReferenceNumber));
+		}
+		if (!dischargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleDischargingMovements(dischargeMovements, dischargeReferenceNumber));
+		}
+		String status = InventoryStatus.done.toString();
+		inventory.setStatus(status);
+		this.updateMedicalInventory(inventory, false);
+		return insertedMovements;
+	}
+
+	private void checkReference(MedicalInventory medicalInventory) throws OHServiceException {
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		String reference = medicalInventory.getInventoryReference();
+		String chargeReferenceNumber = reference + "-charge";
+		String dischargeReferenceNumber = reference + "-discharge";
+		boolean existWithSuffixCharge = movStockInsertingManager.refNoExists(chargeReferenceNumber);
+		boolean existWithSuffixDischarge = movStockInsertingManager.refNoExists(dischargeReferenceNumber);
+		MedicalInventory inventory = this.getInventoryByReference(reference);
+		if (existWithSuffixCharge || existWithSuffixDischarge || (inventory != null && inventory.getId() != medicalInventory.getId())) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.referencealreadyused.msg")));
+		}
+		if (!errors.isEmpty()) {
+			throw new OHServiceException(errors);
+		}
+	}
+
+	/**
+	 * Actualize the {@link MedicalInventory}.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @return {@link MedicalInventory}. It could be {@code null}.
+	 * @throws OHDataValidationException
+	 */
+	public MedicalInventory actualizeMedicalInventoryRow(MedicalInventory inventory) throws OHServiceException {
+		LocalDateTime movFrom = inventory.getInventoryDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		// Fetch all the inventory row of that inventory
+		int id = inventory.getId();
+		List<MedicalInventoryRow> inventoryRowList = medicalInventoryRowManager.getMedicalInventoryRowByInventoryId(id);
+		// TODO: To decide if to make allMedicals parameter
+		boolean allMedicals = true;
+		List<Movement> movs = new ArrayList<>();
+		List<Medical> inventoryMedicalsList = inventoryRowList.stream()
+						.map(MedicalInventoryRow::getMedical)
+						.distinct()
+						.collect(Collectors.toList());
+		if (allMedicals) {
```

Comment:
This condition is always `true` as the variable is set at line 463 and never modified between that assignment and this test.

Reply:
yes, we decided to leave a TODO comment in line 462 because the variable could be controlled by the GUI in the next version after this PR is merged